test/v2/connectors/RetrieveUkPropertyPeriodSummaryConnectorSpec.scala

```diff
@@ -19,101 +19,162 @@ package v2.connectors
+import v2.models.response.retrieveUkPropertyPeriodSummary.{ RetrieveUkPropertyPeriodSummaryResponse, UkFhlProperty, UkNonFhlProperty }
 
 import scala.concurrent.Future
 
 class RetrieveUkPropertyPeriodSummaryConnectorSpec extends ConnectorSpec with ResponseModelsFixture {
 
-  val nino: String = "AA123456A"
-  val businessId: String = "XAIS12345678910"
-  val taxYear: String = "2019-20"
-  val submissionId: String = "4557ecb5-fd32-48cc-81f5-e6acd1099f3c"
-
-  val request: RetrieveUkPropertyPeriodSummaryRequest =
-    RetrieveUkPropertyPeriodSummaryRequest(
-      Nino(nino),
-      businessId,
-      TaxYear.fromMtd(taxYear),
-      submissionId)
-
+  val nino: String                       = "AA123456A"
+  val businessId: String                 = "XAIS12345678910"
+  val submissionId: String               = "4557ecb5-fd32-48cc-81f5-e6acd1099f3c"
   val ukFhlProperty: UkFhlProperty       = UkFhlProperty(None, None)
   val ukNonFhlProperty: UkNonFhlProperty = UkNonFhlProperty(None, None)
 
   def responseWith(ukFhlProperty: Option[UkFhlProperty], ukNonFhlProperty: Option[UkNonFhlProperty]): RetrieveUkPropertyPeriodSummaryResponse =
     RetrieveUkPropertyPeriodSummaryResponse("2020-06-17T10:53:38Z", "2019-01-29", "2020-03-29", ukFhlProperty, ukNonFhlProperty)
 
-  trait Test {
-    _: ConnectorTest =>
+  trait Test extends ConnectorTest {
+
+    def taxYear: String
+    def uri: String
+
     val connector: RetrieveUkPropertyPeriodSummaryConnector = new RetrieveUkPropertyPeriodSummaryConnector(
       http = mockHttpClient,
       appConfig = mockAppConfig
     )
 
-    def stubHttpResponse(outcome: DownstreamOutcome[RetrieveUkPropertyPeriodSummaryResponse])
-    : CallHandler[Future[DownstreamOutcome[RetrieveUkPropertyPeriodSummaryResponse]]]#Derived = {
+    val request: RetrieveUkPropertyPeriodSummaryRequest =
+      RetrieveUkPropertyPeriodSummaryRequest(Nino(nino), businessId, TaxYear.fromMtd(taxYear), submissionId)
+
+    def stubHttpResponse(uri: String, outcome: DownstreamOutcome[RetrieveUkPropertyPeriodSummaryResponse])
+      : CallHandler[Future[DownstreamOutcome[RetrieveUkPropertyPeriodSummaryResponse]]]#Derived = {
       willGet(
-          url = s"$baseUrl/income-tax/business/property/periodic?taxableEntityId=$nino&taxYear=2019-20&incomeSourceId=$businessId&submissionId=$submissionId",
-        )
-        .returns(Future.successful(outcome))
+        url = uri
+      ).returns(Future.successful(outcome))
     }
   }
 
+  trait NonTysTest extends Test with IfsTest {
+    def taxYear: String = "2019-20"
+
+    val uri: String = s"$baseUrl/income-tax/business/property/periodic?taxableEntityId=" +
+      s"$nino&taxYear=2019-20&incomeSourceId=$businessId&submissionId=$submissionId"
+  }
+
+  trait TysTest extends Test with TysIfsTest {
+    def taxYear: String = "2023-24"
+
+    val uri: String =
+      s"$baseUrl/income-tax/business/property/${TaxYear.fromMtd(taxYear).asTysDownstream}/$nino/$businessId/periodic/$submissionId"
```

Comment:
Can we hard code the taxYear into the URL to ensure that we are testing the TYS endpoint